lib/sentry/envelope.ex

```diff
@@ -34,6 +34,41 @@ defmodule Sentry.Envelope do
     }
   end
 
+  @doc """
+  Creates a new envelope containing the client report.
+  since: "10.8.0"
+  """
+  @spec from_client_report(ClientReport.t()) :: t()
+  def from_client_report(%ClientReport{} = client_report) do
+    %__MODULE__{
+      event_id: UUID.uuid4_hex(),
+      items: [client_report]
+    }
+  end
+
+  def get_data_category(type) do
+    if is_binary(type) do
+      type
+    else
+      case type do
+        %Attachment{} ->
+          "attachment"
+
+        %CheckIn{} ->
+          "monitor"
+
+        %ClientReport{} ->
+          "internal"
+
+        %Event{} ->
+          "error"
+
+        _ ->
+          "default"
```

Comment:
This is dangerous because I can call `get_data_category(:completely_banana_value)`. Make sure the input is either a string or a struct (per comm
ent above)